MC/config/PWGHF/external/generator/generator_pythia8_embed_hf.C

```diff
@@ -0,0 +1,176 @@
+            break;
+        default:
+            LOG(fatal) << "********** [GeneratorPythia8EmbedHF] bad configuration, fix it! **********";
+            break;
+        }
+        mGeneratorEvHF->Init();
+    }
+
+    // This function is called by the primary generator
+    // for each event in case we are in embedding mode.
+    // We use it to setup the number of signal events
+    // to be generated and to be embedded on the background.
+    void notifyEmbedding(const o2::dataformats::MCEventHeader* bkgHeader) override
+    {
+        LOG(info) << "[notifyEmbedding] ----- Function called";
+        
+        /// Impact parameter between the two nuclei
+        const float x = bkgHeader->GetB();
+        LOG(info) << "[notifyEmbedding] ----- Collision impact parameter: " << x;
+
+        /// number of events to be embedded in a background event
+        mNumSigEvs = std::max(1.,120.*(x<5.)+80.*(1.-x/20.)*(x>5.)*(x<11.)+240.*(1.-x/13.)*(x>11.));
+        LOG(info) << "[notifyEmbedding] ----- generating " << mNumSigEvs << " signal events " << std::endl;
+    };
+
+protected:
+
+Bool_t importParticles() override
+{
+  /// import particles from underlying event
+  GeneratorPythia8::importParticles();
+  
+  /// Generate mNumSigEvs HF events to be merged in one
+  int nEvsHF = 0;
+  while(nEvsHF < mNumSigEvs) {
+
+      /// generate the HF event
+      bool genOk = false;
+      while(!genOk) {
+          genOk = (mGeneratorEvHF->generateEvent() && mGeneratorEvHF->importParticles() /*copy particles from mGeneratorEvHF.mPythia.event to mGeneratorEvHF.mParticles*/ );
+      }
+
+      /// copy the particles from the HF event in the particle stack
+      auto particlesHfEvent = mGeneratorEvHF->getParticles();
+      int originalSize = mParticles.size(); // stack of this event generator
+      for(int iPart=0; iPart<particlesHfEvent.size(); iPart++) {
+        auto particle = particlesHfEvent.at(iPart);
+
+        /// adjust the particle mother and daughter indices
+        if(particle.GetFirstMother() >= 0)   particle.SetFirstMother(particle.GetFirstMother() + originalSize);
+	    if(particle.GetFirstDaughter() >= 0) particle.SetFirstDaughter(particle.GetFirstDaughter() + originalSize);
+	    if(particle.GetLastDaughter() >= 0)  particle.SetLastDaughter(particle.GetLastDaughter() + originalSize);
+
+        /// copy inside this.mParticles from mGeneratorEvHF.mParticles, i.e. the particles generated in mGeneratorEvHF
+        mParticles.push_back(particle);
+      }
+
+      /// one more event generated, let's update the counter and clear it, to allow the next generation
+      nEvsHF++;
+      //mGeneratedEvents++;
+      mGeneratorEvHF->clearParticles();
+  }
+
+  return true;
+}
+
+private:
+
+    Generator* mGeneratorEvHF; // to generate HF signal events
+    
+    int mNumSigEvs; // number of HF signal events to be merged in one Pythia event
```

Comment:
I would put a default value different from `0`, since I am not sure that the test works properly otherwise.
```suggestion
    int mNumSigEvs{1}; // number of HF signal events to be merged in one Pythia event
```